src/App.tsx

```diff
@@ -14,11 +17,86 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  normalizeDisplayGoods = () => {
+    const { selectedGoods } = this.state;
+    const lengthOfGoods = selectedGoods.length;
+    const magicWord = 'selected';
+
+    switch (lengthOfGoods) {
+      case 0:
+        return `No goods ${magicWord}`;
+      case 1:
+        return `${selectedGoods[0]} is ${magicWord}`;
+      case 2:
+        return `${selectedGoods[0]} and ${selectedGoods[1]} are ${magicWord}`;
+      default:
+        return `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods.slice(-1)} are ${magicWord}`;
+    }
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          {this.normalizeDisplayGoods()}
+        </h1>
+
+        <button
+          className={
+            classNames('button',
+              { hidden: selectedGoods.length === 0 })
+          }
```

Comment:
Let's use more convenient formatting, make sure to fix it everywhere
```suggestion
          className={classNames(
              'button',
              { hidden: selectedGoods.length === 0 }
          )}
```